starknet_devnet/blueprints/rpc/call.py

```diff
@@ -23,6 +23,9 @@ async def call(request: FunctionCall, block_id: BlockId) -> List[Felt]:
         raise RpcError(code=20, message="Contract not found")
 
     try:
+        for calldata in request["calldata"]:
+            if not isinstance(calldata, str):
```

Comment:
- Why not check if it's a felt and not just a string (as per the specifications: https://github.com/starkware-libs/starknet-specs/blob/v0.1.0/api/starknet_api_openrpc.json#L1442)
The logic could be extracted to a function (you could apply `hex` and catch `ValueError`).

- I don't like the fact that `calldata` is used as the name of an item of `request["calldata"]`.

- Is this mentioned in the PR description?

- Looking at `test_call_raises_on_invalid_calldata`, looks like this was already covered, right? So I guess the motivation for introducing this extra check is just to detect bad input earlier?

Reply:
> Why not check if it's a felt and not just a string (as per the specifications: https://github.com/starkware-libs/starknet-specs/blob/v0.1.0/api/starknet_api_openrpc.json#L1442)
The logic could be extracted to a function (you could apply hex and catch ValueError).

i think this is good idea

> I don't like the fact that calldata is used as the name of an item of request["calldata"].
i can change this

> Looking at test_call_raises_on_invalid_calldata, looks like this was already covered, right? So I guess the motivation for introducing this extra check is just to detect bad input earlier?

yes this is for this point

> Is this mentioned in the PR description?
nope but this throws error after i changed the deploy contract

on it!

Reply:
> actually no. so do i need to remove this?

Just update the PR description

Reply:
alrite!

Reply:
done!